static/fetch_lib.sh

```diff
@@ -3,30 +3,23 @@
 true
 # see https://github.com/koalaman/shellcheck/wiki/Directive
 
-# Functions
-msg_box() {
-    [ -n "$2" ] && local SUBTITLE=" - $2"
-    whiptail --title "$TITLE$SUBTITLE" --msgbox "$1" "$WT_HEIGHT" "$WT_WIDTH"
-}
-
-
 IRed='\e[0;91m'         # Red
 IGreen='\e[0;92m'       # Green
 ICyan='\e[0;96m'        # Cyan
 Color_Off='\e[0m'       # Text Reset
 print_text_in_color() {
-        printf "%b%s%b\n" "$1" "$2" "$Color_Off"
+    printf "%b%s%b\n" "$1" "$2" "$Color_Off"
 }
 
 mkdir -p /var/scripts
 if ! [ -f /var/scripts/lib.sh ]
 then
     if ! curl -sfL https://raw.githubusercontent.com/nextcloud/vm/master/lib.sh -o /var/scripts/lib.sh
     then
-        msg_box "You don't seem to have an internet connection and the local lib isn't available. Hence you cannot run this script."
+        print_text_in_color "$IRed" "You don't seem to have an internet connection and the local lib isn't available. Hence you cannot run this script."
         exit 1
     fi
-else
+elif test "$(find /var/scripts/lib.sh -mmin +30)"
```

Comment:
5. we want to cache the file or try to download a new one each time?

Reply:
We managed fine without for the past 6 years, so what are the benefits?

Reply:
one advantage would be, that we don't print out "Updating lib..." every time we run any script.
(Could be a bit annoying when you run many scripts)
Also, we maybe speed up the process a bit, if we don't download a new lib each time we run a script. (could be an advantage on a slow network)

Reply:
Printing "Updating lib..." is a good thing. On slow networks the cursor just stands there an blinks otherwise and the user don't know what's going on.

I feels like untested territory for me, and I know what worked in the past, and I don't know what side effects caching might have on all the scripts in this repo. Sometimes the lib file needs to be 1 second old for stuff to work. I can't tell you exactly where right now, but I know I've coded it such in some places.

Reply:
Generally, I think caching just cause more issues than damage. As late as yesterday I had a customer that said stuff didn't work one day, and the next day all was good. Caching, no thanks.

Reply:
Okay, we might leave it then as it is.
(I'll remove that change)

Reply:
Done